api/src/services/application-csv-export.service.ts

```diff
@@ -60,12 +66,18 @@ export class ApplicationCsvExporterService
   ): Promise<StreamableFile> {
     const user = mapTo(User, req['user']);
     await this.authorizeCSVExport(user, queryParams.listingId);
+
     const filename = join(
       process.cwd(),
       `src/temp/listing-${queryParams.listingId}-applications-${
         user.id
       }-${new Date().getTime()}.csv`,
     );
+
+    if (queryParams.timeZone) {
+      this.timeZone = queryParams.timeZone;
```

Comment:
I don't think this'll work in a complex enough hosted env. We should just pass the incoming timeZone param or the default into the functions to get the timezone where it needs to be in the formatting function


explanation: In NestJS each service is only instantiated 1 time then injected into whichever modules need it. So there is basically 1 application exporter service per dyno right now. I think we might run into problems with the async nature of the exporting functions and the class variable possibly changing regularly 

I would rather us side step the possibility of race conditions affecting this that would be hard to track down, and instead make this as atomic as possible

Hope that makes some sense, if not lets sync up

Reply:
I am okay passing the timezone around as a param. Would just like to note that I followed how we were doing this in the listing-csv-export.service.ts, so we may want to change that in the future as well.